### PR TITLE
Finish Space project

### PR DESCRIPTION
**Pull request by SekoViper**
# Summary

✔️Redux: Fetch data and update the Redux store
✔️Upon the first render fetch data from the SpaceX API endpoints:

APIs
[Rockets](https://api.spacexdata.com/v3/rockets)
[Dragons](https://api.spacexdata.com/v3/dragons)
[Missions](https://api.spacexdata.com/v3/missions)

Highlights
✔️ Created React Project
✔️ Setup Kanban Board
✔️ Set up Empty header
✔️ Implemented Navigation using react-router
✔️ Styled the Header
✔️ Create the basic structure rockets/dragons setup.
✔️ Fetch rocket/dragons data from the API source.
✔️ Display rockets/dragons on the homepage.
✔️ Use redux store to store rockets/dragons data.
✔️Add Remove "Reserved" badge and a "Cancel reservation"
✔️Get Mission Data from the server.
✔️Add Mission Data to redux store.
✔️ Add Style for Joining and leaving Missions Actions
✔️Implemented Joining & leaving Missions
✔️ Display Opt missions in Profile
✔️ Add CSS to Mission Table

### CODE REVIEW

File: src/Components/Dragons/DisplayDragons.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    if (dragons.length === 0) {
+      dispatch(fetchDragonApi());
+    }
+  }, [dispatch]);
+
+  return (
+    <div>
+      <div>
+        {dragons.map((dragon) => (
+          // eslint-disable-next-line react/jsx-props-no-spreading
```
Comment (emmiiorji) [inline src/Components/Dragons/DisplayDragons.jsx]: Linters represent coding styles that you share with your team. As much as possible, kindly avoid disabling them in line as they most of the time also enforce best coding practices.

File: src/Components/Dragons/Dragons.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { toggleReserved } from '../../Redux/dragonSlice';
+
+/* eslint-disable react/prop-types */
```
Comment (emmiiorji) [inline src/Components/Dragons/Dragons.jsx]: As mentioned above, linters represent coding styles that you share with your team. As much as possible, kindly avoid disabling them in line as they most of the time also enforce best coding practices.

_There are about 10 other occurrences, kindly remove them too and fix the errors that result._

File: .gitignore [hunk 3]
```diff
@@ -4,6 +4,7 @@
 /node_modules
 /.pnp
 .pnp.js
+todo.txt
```
Comment (emmiiorji) [inline .gitignore:7]: [Optional] You seem to have attempted to ignore this file after committing it. You may want to delete it, commit the change and create it again to be properly ignored.

File: src/Components/Missions/missionsPage.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      dispatch(getMissionsData());
+    }
+  }, [dispatch], missions.missions.length);
+
+  return (
+    missions.status === 'loading' ? <h2>Loading</h2> : <MissionsList missions={missions.missions} />
+  );
+};
+
+export default MissionsPage;
```
Comment (emmiiorji) [inline src/Components/Missions/missionsPage.js]: As stated in the [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#note-make-sure-to-add-test-for-all-your-components-and-functions) for the project, kindly write unit tests with React Testing Library and Jest (for unit tests).

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @SekoViper, @studiosnabeel and @MosDevx, 
# Status: Changes Required ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

A little bit more

## Good Highlights✔️

- Linter checks are passing
- Adhered to Gitflow
- Good commit messages
- Good design


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @studiosnabeel @SekoViper @MosDevx , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**